Tools/CMake/AMReXThirdPartyLibraries.cmake

```
@@ -3,13 +3,13 @@
 #
 if (ENABLE_HDF5)
     set(HDF5_PREFER_PARALLEL TRUE)
-    find_package(HDF5 1.10.4 REQUIRED COMPONENTS CXX)
+    find_package(HDF5 1.10.4 REQUIRED)
     if (NOT HDF5_IS_PARALLEL)
         message(FATAL_ERROR "\nHDF5 library does not support parallel I/O")
     endif ()
-    target_include_directories(amrex PUBLIC ${HDF5_CXX_INCLUDE_DIRS})
-    target_compile_definitions(amrex PUBLIC ${HDF5_CXX_DEFINES})
-    target_link_libraries(amrex PUBLIC ${HDF5_CXX_LIBRARIES})
+    target_include_directories(amrex PUBLIC ${HDF5_C_INCLUDE_DIRS})
+    target_compile_definitions(amrex PUBLIC ${HDF5_C_DEFINES})
+    target_link_libraries(amrex PUBLIC ${HDF5_C_LIBRARIES})
```

Ok, but this does as well and is what most people use (so it might be better tested in the end):
```suggestion
    target_include_directories(amrex PUBLIC ${HDF5_INCLUDE_DIRS})
    target_compile_definitions(amrex PUBLIC ${HDF5_DEFINES})
    target_link_libraries(amrex PUBLIC ${HDF5_LIBRARIES})
```

---

There is no indication in the doc that one form is preferred over the other, therefore I am inclined to keep the current form with the the language specified in the variable name.

---

In that case I recommend to request `C` explicitly, because you use a component on which you rely with default behavior.

The review I provide here stems from experience, e.g. if internal dependencies or build variants change and introduce publicly undeclared internal side-dependencies.